File: compiler/cfg.rb
#every directory is relative to the compile.rb file, which should be located in projects root
#source files location
$groups = './source/groups/'            #contains .grp and .opt files
$modifiers = './source/modifiers.grp'   #a single file containing modifiers buttons, default: modifiers.grp
$prefabs = './source/prefabs/'
#a directory to store build files
$build = './build/'
#output folder
$out_dir = './output/'
#compiler result location
$framework = $out_dir + 'framework.cfg'
$reset = $out_dir + 'resets.cfg'
$group = $out_dir + 'groups.cfg'
$modifier = $out_dir + 'modifiers.cfg'
#directory of compiler parts

File: compiler/cfg.py
#! python3


class CFG:
    groups = './source/groups/'                 # source files for groups
    modifiers = './source/modifiers.grp'       # surce file for modifiers
    prefabs = './source/prefabs/'               # source files for prefabs
    build = './build/'                           # directory to store temp files
    out_dir = './output/'                        # final output directory
    # file to store base definitions in
    oFramework = out_dir+'/framework.cfg'
    # file to store reset commands in
    # usefull when reapplying framework
    oResets = out_dir+'/resets.cfg'
    # file to store group definitions in
    oGroups = out_dir+'/groups.cfg'
    # file to store modifiers' definitions in
    oModifiers = out_dir+'/modifiers.cfg'
